csrc/mmdeploy/codebase/mmseg/segment.cpp

```diff
@@ -41,40 +45,60 @@ class ResizeMask : public MMSegmentation {
                      mask.shape());
       return Status(eNotSupported);
     }
-    if (!with_argmax_) {
-      MMDEPLOY_ERROR("TODO: SDK will support probability featmap soon.");
+    if ((mask.data_type() != DataType::kFLOAT) && !with_argmax_) {
+      MMDEPLOY_ERROR("probability feat map only support float32 output");
       return Status(eNotSupported);
     }
 
+    auto channel = (int)mask.shape(1);
     auto height = (int)mask.shape(2);
     auto width = (int)mask.shape(3);
     auto input_height = preprocess_result["img_metas"]["ori_shape"][1].get<int>();
     auto input_width = preprocess_result["img_metas"]["ori_shape"][2].get<int>();
+    if (!with_argmax_) {
+      // (C, H, W) -> (H, W, C)
+      ::mmdeploy::operation::Context ctx(device_, stream_);
+      std::vector<int> axes = {0, 2, 3, 1};
+      permute_.Apply(mask, mask, axes);
+    }
     Device host{"cpu"};
     OUTCOME_TRY(auto host_tensor, MakeAvailableOnDevice(mask, host, stream_));
     OUTCOME_TRY(stream_.Wait());
 
-    OUTCOME_TRY(auto cv_type, GetCvType(mask.data_type()));
+    OUTCOME_TRY(auto cv_type, GetCvType(mask.data_type(), channel));
     cv::Mat mask_mat(height, width, cv_type, host_tensor.data());
 
-    if (mask_mat.channels() > 1) {
-      cv::extractChannel(mask_mat, mask_mat, little_endian_ ? 0 : mask_mat.channels() - 1);
-    }
-    if (mask_mat.type() != CV_32S) {
-      mask_mat.convertTo(mask_mat, CV_32S);
-    }
+    cv::Mat resized_mask;
+    cv::Mat resized_score;
+
+    Tensor tensor_mask{};
+    Tensor tensor_score{};
 
-    cv::Mat resized_mask = cpu::Resize(mask_mat, input_height, input_width, "nearest");
+    if (with_argmax_) {
+      // mask
+      if (mask_mat.channels() > 1) {
+        cv::extractChannel(mask_mat, mask_mat, little_endian_ ? 0 : mask_mat.channels() - 1);
+      }
+      if (mask_mat.type() != CV_32S) {
+        mask_mat.convertTo(mask_mat, CV_32S);
+      }
+      resized_mask = cpu::Resize(mask_mat, input_height, input_width, "nearest");
+      tensor_mask = cpu::CVMat2Tensor(resized_mask);
+    } else {
+      // score
```

Comment:
can you take a look [1648](https://github.com/open-mmlab/mmdeploy/issues/1648)

I google this error, someone said that cv::resize does not support int32, so I changed the code of 0.12.0 to the following, it seems to work well, but I haven't tried the code of this branch, because I think it can't solve my problem
```
    if (mask_mat.type() != CV_32F) {
      mask_mat.convertTo(mask_mat, CV_32F);
    }

    cv::Mat resized_mask = cpu::Resize(mask_mat, input_height, input_width, "nearest");
```
Another thing is strange, the resize of CV_32S on 0.10.0 is correct, I compared the input before resize of the two versions, they are the same, 
```
[2023-01-13 17:49:00.016] [mmdeploy] [info] [segment.cpp:33] tensor.name: output, tensor.shape: [1, 1, 512, 512], tensor.data_type: INT32
```

but the version of ppl.cv used by 0.10.0 and 0.12.0 is also the same, I can't find the original cause of my problem.
I will keep trying to find the reason, if you have any suggestions, or this branch has been able to solve my problem，please let me know

Reply:
Thank you for your patience，I saw that it uses opencv, but I only noticed the version of opencv, and it was installed through ppl.cv, I didn't notice the call parameter of resize，Thanks again，With your guidance, I have solved my problem